auslib/db.py

```diff
@@ -1785,20 +1794,21 @@ def localeExists(self, name, platform, locale, transaction=None):
 
     def delete(self, where, changed_by, old_data_version, transaction=None, dryrun=False):
         names = []
-        mapping_count = dbo.rules.t.count().where(dbo.rules.mapping == where["name"]).execute().fetchone()[0]
-
-        whitelist_count = dbo.rules.t.count().where(dbo.rules.whitelist == where["name"]).execute().fetchone()[0]
-
-        fallbackMapping_count = dbo.rules.t.count().where(dbo.rules.fallbackMapping == where["name"]).execute().fetchone()[0]
-
-        if mapping_count > 0 or whitelist_count > 0 or fallbackMapping_count > 0:
-            msg = "%s has rules pointing to it. Hence it cannot be deleted." % (self.name)
-            raise ValueError(msg)
         for toDelete in self.select(where=where, columns=[self.name, self.product], transaction=transaction):
             names.append(toDelete["name"])
             self._proceedIfNotReadOnly(toDelete["name"], transaction=transaction)
             if not self.db.hasPermission(changed_by, "release", "delete", toDelete["product"], transaction):
                 raise PermissionDeniedError("%s is not allowed to delete releases for product %s" % (changed_by, toDelete["product"]))
+
+        for name in names:
```

Comment:
We always sort-of did at the database layer. Prior to Scheduled Changes, you could've called Releases.delete() to do it (because we didn't override it at that time). One of the side effects of the Scheduled Changes implementation was overriding methods such as this, which required changing their interface to match the base class, which in turn means we have methods like this that allow deleting anything any number of things at once.

I agree that this feels more dangerous when it's being down in this specific class. Perhaps it's best to bail out early in this method if "name" isn't in "where", or if the clause matches more than one row?